File: Task/Table-creation-Postal-addresses/Clojure/table-creation-postal-addresses.clj
(require '[clojure.java.jdbc :as sql])
; Using h2database for this simple example.
(def db {:classname "org.h2.Driver"
         :subprotocol "h2:file"
         :subname "db/my-dbname"})

(sql/db-do-commands db
  (sql/create-table-ddl :address
    [:id "bigint primary key auto_increment"]
    [:street "varchar"]
    [:city "varchar"]
    [:state "varchar"]
    [:zip "varchar"]))

File: Task/Table-creation-Postal-addresses/Lua/table-creation-postal-addresses.lua
-- Import module
local sql = require("ljsqlite3")

-- Open connection to database file
local conn = sql.open("address.sqlite")

-- Create address table unless it already exists
conn:exec[[
CREATE TABLE IF NOT EXISTS address(
  id INTEGER PRIMARY KEY AUTOINCREMENT,
  street TEXT NOT NULL,
  city TEXT NOT NULL,
  state TEXT NOT NULL,
  zip TEXT NOT NULL)
]]

-- Explicitly close connection
conn:close()
